user-input.js

```diff
@@ -0,0 +1,87 @@
+const fs = require('fs');
+const prompt = require('prompt');
+const colors = require('colors');
+const utilities = require('./utilities');
+const api = require('./api');
+
+function buildPost(dir, files) {
+  const post = {
+    description: utilities.getDescription(),
+    public: true,
+    files: {},
+  };
+
+  let filesProcessed = 0;
+
+  files.forEach((file, index, array) => {
+    fs.readFile(`${dir}${file}`, 'utf8', (error, content) => {
+      if (error) throw error;
+      post.files[file] = {
+        content,
+      };
+
+      filesProcessed++;
+      if (filesProcessed === array.length) {
+        api.postGist(post);
+      }
+    });
+  });
+}
+
+function readFile(name) {
+  buildPost(`${process.cwd()}/`, [name]);
+}
+
+function readDirectory(dirName) {
+  const fullDir = `${process.cwd()}/${dirName}/`;
+  fs.readdir(fullDir, (error, files) => {
+    if (error) console.log(error.message.red);
+    buildPost(fullDir, files);
+  });
+}
+
+function valueFromPrompt(option, promptText) {
+  prompt.message = '';
+  prompt.start();
+
+  return prompt.get({
+    properties: {
+      value: {
+        description: promptText,
+      },
+    },
+  }, (error, result) => {
+    if (error) console.log(error.message.red);
+    if (option === 1) readFile(result.value);
+    if (option === 2) readDirectory(result.value);
+  });
+}
+
+function showMenu() {
```

Comment:
perhaps rather than `showMenu` UI, let's try taking in a path as an argument and parsing that using the file system, i.e.:

`$ gist-uploader ~/documents/filename.txt`